### PR TITLE
Add some docs for Queue and tests

### PR DESCRIPTION
**Pull request by nomisRev**
Some improvements based on the work today with @danimontoya on grpc Kotlin.

### CODE REVIEW

File: arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/concurrent/QueueTest.kt [hunk 1]
```diff
@@ -188,4 +192,94 @@ class QueueTest : StreamSpec(spec = {
...
+        .concurrently(
+          s.noneTerminate()
+            .through(q.enqueue())
+        ).toList() shouldBe s.toList()
+    }
+  }
+
+  "Queue.bounded(0) with outstanding taker cannot offer" {
+    val q = Queue.bounded<Int>(0)
+    ForkConnected { q.dequeue1() }
```
Comment (danimontoya) [inline arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/concurrent/QueueTest.kt:256]: Is this needed?

**Review by danimontoya — APPROVED** [review]
Thank you @nomisRev !!! 👍

File: arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/concurrent/QueueTest.kt (hunk 1 above)
Comment (nomisRev) [reply arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/concurrent/QueueTest.kt:256]: That is to test that even tho someone is trying to take from the `Queue`, that `offer` still fails. We can't assume from the test that it won't do that, right? 😅

**Review by aballano — APPROVED** [review]
👌

**Pull request merged**